lib/neo4j/shared/enum.rb

```diff
@@ -0,0 +1,120 @@
+require 'active_support/core_ext/hash/except'
+require 'active_support/core_ext/hash/slice'
+
+module Neo4j::Shared
+  module Enum
+    extend ActiveSupport::Concern
+
+    class ConflictingEnumMethodError < Neo4j::Error; end
+
+    module ClassMethods
+      attr_reader :neo4j_enum_data
+
+      def enum(parameters = {})
+        options, parameters = *split_options_and_parameters(parameters)
```

Comment:
Could we just have all of the parameter be all of the arguments except for an optional last options argument?  ActiveSupport even has a (not very complicated) method for this:

http://api.rubyonrails.org/v2.3/classes/ActiveSupport/CoreExtensions/Array/ExtractOptions.html

For the options, it would be nice if an `ArgumentError` was raised if an option was given that wasn't in `VALID_OPTIONS_FOR_ENUMS`